### PR TITLE
Support for key aliases, key handling tests

### PR DESCRIPTION
**Pull request by darrenburns**
- Adds support for key aliases.
- Adds message pump unit tests for dispatching keys and ensuring handlers are called as expected.

If multiple handlers exist for a single key, an exception is raised since we can't know which one the author meant.

Also adds the exception logic to the case where public and private handlers exist for the same key press.

Example output for conflicting handlers:

image

### CODE REVIEW

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: src/textual/message.py [hunk 1]
```diff
@@ -113,7 +114,7 @@ def stop(self, stop: bool = True) -> Message:
         self._stop_propagation = stop
         return self
 
-    async def _bubble_to(self, widget: Widget) -> None:
+    async def bubble_to(self, widget: MessagePump) -> None:
```
[the remark this answers is not in the document]
Comment (darrenburns) [reply src/textual/message.py:117]: It's used outside of this class

[the remark this answers is not in the document]
Comment (willmcgugan) [reply src/textual/message.py:117]: I think for our purposes, anything that a Textual app developer doesn't need to use should be private. It also prevents the method from being exposed in the docs.

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Looks good. Just one request.

**Comment by willmcgugan** [comment]
@darrenburns Can be drop the property now?

**Comment by darrenburns** [comment]
@willmcgugan Yep, sorry, missed that.

**Review by willmcgugan — APPROVED** [review]
LGTM

**Pull request merged**